packages/file_selector/file_selector_platform_interface/lib/src/types/x_file/html.dart

```diff
@@ -0,0 +1,161 @@
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'package:http/http.dart' as http show readBytes;
+import 'package:meta/meta.dart';
+import 'dart:html';
+
+import './base.dart';
+
+/// A XFile that works on web.
+///
+/// It wraps the bytes of a selected file.
+class XFile extends XFileBase {
+  String path;
+
+  final String mimeType;
+  final Uint8List _data;
+  final int _length;
+  final String name;
+  final DateTime _lastModified;
+  Element _target;
+
+  final XFileTestOverrides _overrides;
+
+  bool get _hasTestOverrides => _overrides != null;
+
+  /// Construct a XFile object from its ObjectUrl.
+  ///
+  /// Optionally, this can be initialized with `bytes` and `length`
+  /// so no http requests are performed to retrieve files later.
+  ///
+  /// `name` needs to be passed from the outside, since we only have
+  /// access to it while we create the ObjectUrl.
+  XFile(
+      this.path, {
+        this.mimeType,
+        this.name,
+        int length,
+        Uint8List bytes,
+        DateTime lastModified,
+        @visibleForTesting XFileTestOverrides overrides,
+      })  : _data = bytes,
+        _length = length,
+        _overrides = overrides,
+        _lastModified = lastModified,
+        super(path);
+
+  /// Construct an XFile from its data
+  XFile.fromData(
+      Uint8List bytes, {
+        this.mimeType,
+        this.name,
+        int length,
+        DateTime lastModified,
+        @visibleForTesting XFileTestOverrides overrides,
+      })  : _data = bytes,
+        _length = length,
+        _overrides = overrides,
+        _lastModified = lastModified,
+        super('') {
+    Blob blob;
+    if (mimeType == null) {
+      blob = Blob([bytes]);
+    } else {
+      blob = Blob([bytes], mimeType);
+    }
+    this.path = Url.createObjectUrl(blob);
+  }
+
+
+
+  Future<Uint8List> get _bytes async {
+    if (_data != null) {
+      return Future.value(UnmodifiableUint8ListView(_data));
+    }
+    return http.readBytes(path);
+  }
+
+  @override
+  Future<int> length() async {
+    return _length ?? (await _bytes).length;
+  }
+
+  @override
+  Future<String> readAsString({Encoding encoding = utf8}) async {
+    return encoding.decode(await _bytes);
+  }
+
+  @override
+  Future<Uint8List> readAsBytes() async {
+    return Future.value(await _bytes);
+  }
+
+  @override
+  Stream<Uint8List> openRead([int start, int end]) async* {
+    final bytes = await _bytes;
+    yield bytes.sublist(start ?? 0, end ?? bytes.length);
+  }
+
+  /// Saves the data of this XFile at the location indicated by path.
+  /// For the web implementation, the path variable is ignored.
+  void saveTo(String path) async {
+    // Create a DOM container where we can host the anchor.
+    _target = _ensureInitialized('__x_file_dom_element');
```

Comment:
This looks very similar to the `load` code as well, some more reason to split it (and maybe reuse it)